pyiceberg/table/__init__.py

```diff
@@ -1129,12 +1128,22 @@ def _(update: SetSnapshotRefUpdate, base_metadata: TableMetadata, context: _Tabl
     return base_metadata.model_copy(update=metadata_updates)
 
 
+@_apply_table_update.register(RemoveSnapshotRefUpdate)
+def _(update: RemoveSnapshotRefUpdate, base_metadata: TableMetadata, context: _TableMetadataUpdateContext) -> TableMetadata:
```

Comment:
Hi @kevinjqliu, I've already implemented this function and relevant Remove Snapshot apis in #822, which is waiting on #758. 

If you need to add this functionality urgently, please feel free to use the code in the linked PRs.

Reply:
Cool! Thanks for the heads up. I can rebase once those PRs are in.

Reply:
Do you think it'll be helpful to enforce any new TableUpdate class to have a corresponding update function? 
Like in #952

Reply:
I think that would helpful! In case a feature is not implemented, like `RemoveSnapshotRefUpdate` and `RemoveSnapshotsUpdate`, the test should atleast print a message saying so.
It would help keep track of which features are / are not implemented, and won't be surprise to the end user or us.